pkgs/by-name/ec/ecc/package.nix

```diff
@@ -0,0 +1,115 @@
+{ lib
+, stdenv
+, makeWrapper
+, fetchFromGitHub
+, rustPlatform
+, pkg-config
+, llvmPackages_16
+, elfutils
+, zlib
+}:
+let
+  bpftool = stdenv.mkDerivation {
+    pname = "bpftool";
+    version = "unstable-2023-03-11";
+
+    # this fork specialized for some functions
+    # and has eventually been embedded into the ecc binary
+    src = fetchFromGitHub {
+      owner = "eunomia-bpf";
+      repo = "bpftool";
+      rev = "05940344f5db18d0cb1bc1c42e628f132bc93123";
+      hash = "sha256-g2gjixfuGwVnFlqCMGLWVPbtKOSpQI+vZwIZciXFPTc=";
+      fetchSubmodules = true;
+    };
+
+    buildInputs = [ llvmPackages_16.clang llvmPackages_16.libllvm.dev elfutils zlib ];
```

Comment:
@oluceps sorry for all of those iterations, but I was wondering now. Why is `clang` here? Could it be it belongs to `nativeBuildInputs`? And why do you explicitly specify version 16 here? I'm guessing you wanted the same llvm as used by rust? I have tested the changes I just pushed to this PR, and it seems fine. Tell me what do you think, and if you like it. Please squash my commits.

Reply:
> @oluceps sorry for all of those iterations, but I was wondering now. Why is `clang` here? Could it be it belongs to `nativeBuildInputs`? And why do you explicitly specify version 16 here? I'm guessing you wanted the same llvm as used by rust? I have tested the changes I just pushed to this PR, and it seems fine. Tell me what do you think, and if you like it. Please squash my commits.

The default llvm version in nixpkgs is also works, but I think keeping it identical with rust's llvm version is indeed better. `clang` in here is to enable clang `CO-RE` feature for `bpftool`. Thanks for your helpful insights!